src/string/extglobmatch.c

```diff
@@ -0,0 +1,459 @@
+                }
+                break;
+            case '*':
+                /* Matches zero or more occurences of the given patterns */
+                if (con.pat[0] == '(')
+                {
+                    struct ext_glob eg = find_next((struct ext_glob)
+                                                        {.cur_pat = con.pat});
+                    int res;
+                    while (!eg.ret) {
+                        struct context arg = {  .str = con.str,
+                                                .pat = eg.cur_pat,
+                                                .pat_chunk_end = eg.end_pat,
+                                                .back_pat = back_pat,
+                                                .back_str = back_str,
+                                                .repeat = 1,
+                                                .repeat_pat = eg.cur_pat,
+                                                .nested = con.nested + 1};
+                        res = glob_match(arg);
+                        if (res == 1) {
+                            return res;
+                        }
+                        eg = find_next(eg);
+                    }
+                    con.pat = eg.end_pat;
+                } else { /* Any-length wildcard */
+                    if  (*con.pat == '\0') { /* Optimize trailing * case */
+                        return 1;
+                    }
+                    back_pat = con.pat;
+                    back_str = con.str;   /* Allow zero-length match */
+                }
+                break;
+            case '+':
+                /* Matches one or more of the given patterns */
+                if (con.pat[0] == '(') {
+                    struct ext_glob eg = find_next((struct ext_glob)
+                                                        {.cur_pat = con.pat});
+                    if (c == '\0') /// TODO check this
+                        return 0;
+                    int res;
+                    while (!eg.ret) {
+                        struct context arg = {  .str = con.str,
+                                                .pat = eg.cur_pat,
+                                                .pat_chunk_end = eg.end_pat,
+                                                .back_pat = back_pat,
+                                                .back_str = back_str,
+                                                .repeat = 1,
+                                                .repeat_pat = eg.cur_pat,
+                                                .nested = con.nested + 1};
+                        res = glob_match(arg);
+                        if (res == 1) {
+                            return res;
+                        }
+                        eg = find_next(eg);
+                    }
+                    return 0;
+                } else {
+                    malformed_pattern(p);
+                }
+                break;
```

Comment:
This line is also unreachable -- either move the return 0 here or replace the break with `/* not reachable */`.